google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineCloudType.java

```diff
@@ -118,8 +118,8 @@ public void connect(@NotNull ConnectionCallback<AppEngineDeploymentConfiguration
 
       if (!Services.getLoginService().isLoggedIn()) {
         callback.errorOccurred(GctBundle.message("appengine.deployment.error.not.logged.in"));
-      } else if (CloudSdkUtil.isCloudSdkExecutable(
-          CloudSdkUtil.toExecutablePath(CloudSdkService.getInstance().getCloudSdkHomePath()))) {
+      } else if (CloudSdkUtil.isCloudSdkExecutable(CloudSdkUtil.toExecutablePath(
+          CloudSdkService.getInstance().getCloudSdkHomePath().toString()))) {
```

Comment:
If it's possible for the cloudSdkHomePath to be invalid why not expose that as a method on the service? e.g. cloudSdkService.hasValidSdk() or something like that.